packages/next/src/server/load-components.ts

```diff
@@ -272,8 +255,18 @@ async function loadComponentsImpl<N = any>({
   const Document = interopDefault(DocumentMod)
   const App = interopDefault(AppMod)
 
-  const { getServerSideProps, getStaticProps, getStaticPaths, routeModule } =
-    ComponentMod
+  const { getServerSideProps, getStaticProps, getStaticPaths } = ComponentMod
+
+  const routeModule = ComponentMod.routeModule as RouteModule
+
+  const reactLoadableManifest =
+    // APP_ROUTE does not have a react-loadable-manifest.
+    routeModule.definition.kind === RouteKind.APP_ROUTE
```

Comment:
looks like vade got this one right and the cast is confusing ts

in the integration tests i see `TypeError: Cannot read properties of undefined (reading 'definition')`
```suggestion
    routeModule?.definition.kind === RouteKind.APP_ROUTE
```

Reply:
The types are wrong though... need to figure out what case this hits

Reply:
Dug into this and turns out the root cause is that we try to load `/_app` as a route, which is incorrect. Still updating all cases as there is a better way that just skipping this